StRoot/StFttClusterMaker/StFttClusterMaker.cxx

```diff
@@ -0,0 +1,429 @@
+            for ( StFttCluster * clu : vClusters ){
+                mFttCollection->addCluster( clu );
+                nClusters++;
+            }
+            auto hdClusters = FindClusters( dhStripsPerRob[iRob] );
+            // Add them to StEvent  
+            for ( StFttCluster * clu : hdClusters ){
+                mFttCollection->addCluster( clu );
+                nClusters++;
+            }
+            auto vdClusters = FindClusters( dvStripsPerRob[iRob] );
+            // Add them to StEvent  
+            for ( StFttCluster * clu : vdClusters ){
+                mFttCollection->addCluster( clu );
+                nClusters++;
+            }
+        } // loop on iRob
+    } // nStripsHit
+    LOG_INFO << "Found " << nClusters << " clusters this event" << endm;
+
+    return kStOk;
+} // Make
+
+void StFttClusterMaker::InjectTestData(){
+    mFttCollection->rawHits().clear();
+
+    StFttRawHit *hit = new StFttRawHit( 1, 1, 1, 1, 1, 55, 1, 1, 0 );
+    hit->setMapping( 1, 1, 1, 23, kFttHorizontal ); // LEFT 2
+    mFttCollection->addRawHit( hit );
+
+    hit = new StFttRawHit( 1, 1, 1, 1, 1, 90, 1, 1, 0 );
+    hit->setMapping( 1, 1, 1, 24, kFttHorizontal ); // LEFT 1
+    mFttCollection->addRawHit( hit );
+
+    hit = new StFttRawHit( 1, 1, 1, 1, 1, 60, 1, 1, 0 );
+    hit->setMapping( 1, 1, 1, 27, kFttHorizontal );
+    mFttCollection->addRawHit( hit );
+
+    hit = new StFttRawHit( 1, 1, 1, 1, 1, 95, 1, 1, 0 );
+    hit->setMapping( 1, 1, 1, 25, kFttHorizontal ); // CENTER
+    mFttCollection->addRawHit( hit );
+
+    hit = new StFttRawHit( 1, 1, 1, 1, 1, 93, 1, 1, 0 );
+    hit->setMapping( 1, 1, 1, 26, kFttHorizontal );
+    mFttCollection->addRawHit( hit );
+
+    hit = new StFttRawHit( 1, 1, 1, 1, 1, 19, 1, 1, 0 );
+    hit->setMapping( 1, 1, 1, 28, kFttHorizontal );
+    mFttCollection->addRawHit( hit );
+} // InjectTestData
+
+
+bool StFttClusterMaker::PassTimeCut( StFttRawHit * hit ){
+    return (abs( hit->time() ) <=3); 
+}
+
+
+StFttRawHit * StFttClusterMaker::FindMaxAdc( std::vector<StFttRawHit *> hits, size_t &pos ){
+    auto itMax = std::max_element(hits.begin(),
+                             hits.end(),
+                             [](const StFttRawHit* a,const StFttRawHit* b) { return a->adc() < b->adc(); });
+
+    pos = (itMax - hits.begin());
+    if ( pos >= hits.size() )
+        return nullptr;
+    return *itMax;
+}
+
+void StFttClusterMaker::SearchClusterEdges( std::vector< StFttRawHit * > hits, 
+                                            size_t start, // start index at MaxADC
+                                            size_t &left, size_t &right ){
+    // set initial values
+    left     = start;
+    right    = start;
+
+    auto lastHitLeft  = hits[start];
+    auto lastHitRight = hits[start];
+
+    bool searchRight = true;
+    bool searchLeft = true;
+
+    StFttRawHit *hitLeft = nullptr, *hitRight = nullptr;
+
+    while ( searchRight || searchLeft ){
+        if ( mDebug ){
```

Comment:
--> LOG_DEBUG

Reply:
All classes which inherit from StMaker have a SetDebug( int debuglvl ) method.  You can retrieve the debug level using the Debug() method.  I believe that the logger refers to the debug level in order to decide whether LOG_DEBUG lines should be written or not.

Reply:
Addressed: 3686a6eea3b300c95180bdcee0ad698b5763a88c